synfig-studio/src/gui/workarea.cpp

```diff
@@ -1031,12 +1031,23 @@ bool
 WorkArea::on_key_release_event(GdkEventKey* event)
 {
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
+	const float radius((std::fabs(pw)+std::fabs(ph))*4);
 	rotate_guide = false;
 	auto event_result = canvas_view->get_smach().process_event(
 		EventKeyboard(EVENT_WORKAREA_KEY_UP, event->keyval, Gdk::ModifierType(event->state)) );
 	if (event_result != Smach::RESULT_OK)
 		return true;
 
+	if (event->keyval == GDK_KEY_Escape && guide_highlighted)
+	{
+		get_guide_list().erase(curr_guide);
+		drawing_area->queue_draw();
+		GuideList::iterator iter = find_guide(get_cursor_pos(), radius);
+		curr_guide = iter;
+		drawing_area->queue_draw();
```

Comment:
1. Why do you need two `queue_draw()` calls in a row? Although it won't make it redraw twice, it has no point.
2. After you remove a guide from `GuideList`, you have to save the metadata, as seen in L1505-1515. Otherwise, `ctrl+Z` won't work.

Reply:
Item 2 I think I fixed in #3304